src/http-tequilapi-client.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2019 The "mysteriumnetwork/js-tequilapi" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { HttpInterface } from './adapters/interface'
+import ProposalsQuery from './adapters/proposals-query'
+import { TequilapiClient } from './client'
+import { ConnectionIPDTO, parseConnectionIPDTO } from './dto/connection-ip'
+import { ConnectionStatisticsDTO, parseConnectionStatisticsDTO } from './dto/connection-statistics'
+import { ConnectionStatusDTO, parseConnectionStatusDTO } from './dto/connection-status-dto'
+import { ConsumerLocationDTO, parseConsumerLocationDTO } from './dto/consumer-location'
+import { parseIdentitiesResponseDTO } from './dto/identities-response'
+import { IdentityDTO, parseIdentityDTO } from './dto/identity'
+import {
+  IdentityRegistrationDTO,
+  parseIdentityRegistrationDTO
+} from './dto/identity-registration/identity-registration'
+import { NodeHealthcheckDTO, parseHealthcheckResponse } from './dto/node-healthcheck'
+import { ProposalDTO } from './dto/proposal'
+import { parseProposalsResponseDTO } from './dto/proposals-response'
+import { ConnectionRequest } from './dto/query/connection-request'
+import { ProposalQueryOptions } from './dto/query/proposals-query-options'
+import { TIMEOUT_DISABLED } from './timeouts'
+
+export class HttpTequilapiClient implements TequilapiClient {
```

Comment:
It's just `TequilapiClient`, bacause we have `HTTPAdapter` implementation which isolates transport type. So client class doest not need to leak information about transport being used under the hood

`this.http` -> `this.adapter`
`this.http` -> `this.transport`

Reply:
Well, `get`/`post`/`put`/`delete` are terms on `HTTP`, so `HttpAdapter` isn't actually generic one.
And this implementation is not a generic one as well - it relies on adapter `get`/`post`/... methods, which basically means it relies on `HTTP`.

Anyways, it's out of scope of this PR, so would suggest to discuss this outside of PR.

Reply:
I would stay `get`, `post` is way to transport resources, just it